clang/include/clang/AST/Availability.h

```diff
@@ -57,6 +58,57 @@ class AvailabilitySpec {
   bool isOtherPlatformSpec() const { return Version.empty(); }
 };
 
+class Decl;
+
+/// Storage of availability attributes for a declaration.
+struct AvailabilityInfo {
+  /// The domain is the platform for which this availability info applies to.
+  llvm::SmallString<32> Domain;
```

Comment:
Are 32 chars required?

Reply:
In fear I may have misunderstood your question, I give alternative answers. 
* There is not a default template argument for SmallString 
* For why I picked `32`, `31` characters is the longest string value under inputs and outputs for [mapping platform availability](https://github.com/llvm/llvm-project/blob/main/clang/include/clang/Basic/Attr.td#L984)